### PR TITLE
Default to system locale language

### LINKED ISSUE #48: Show English in translations first
The first language in the language chooser is chineese, and you have to look in the list to find English. I think it should be on the top as the majority of pepole will use English anyway.

### PR DESCRIPTION
**Pull request by oynqr**
With these changes PolyMC will try to default to the system locale on first startup.

This is honestly a bit hacky, since it's my first time working with QT/this codebase, but I don't think it's much worse than the surrounding code.

Closes #48

### CODE REVIEW

**Comment by ghost** [comment]
I believe people wanted it to have the locale selected as well. If you could do that it'd be great.

Add a "Closes" to this so it automatically closes the related issue.

File: launcher/translations/TranslationsModel.cpp [hunk 1]
```diff
@@ -439,6 +460,12 @@ bool TranslationsModel::selectLanguage(QString key)
 {
     QString &langCode = key;
     auto langPtr = findLanguage(key);
+
+    if (langCode.length() == 0)
```
Comment (ghost) [inline launcher/translations/TranslationsModel.cpp]: just use `langCode.empty()`

**Review by ghost — CHANGES_REQUESTED** [review]

File: launcher/translations/TranslationsModel.cpp (hunk 1 above)
Comment (oynqr) [reply launcher/translations/TranslationsModel.cpp]: Done.

**Comment by oynqr** [comment]
This should address most, if not all, of the issue. If there is a corresponding language to the system locale, it should always be at the top of the list and it will also be highlighted/selected.
Please note that this changes the behavior in case the user terminates PolyMC with the initial language selector still open, as it will no longer open it on subsequent starts since the language config was set to the locale default or American English if there is no translation available.

I have not and can not test this on Mac OS/Windows, so that would be appreciated.

**Comment by ghost** [comment]
Can't test on MacOS but can on Windows sometime later.

**Comment by oynqr** [comment]
I might spin up a macOS VM to test this tomorrow.

**Comment by oynqr** [comment]
Seems like it's hitting this on macOS: https://bugreports.qt.io/browse/QTBUG-72491. Creating the relevant .lproj directories manually fixes it.

**Comment by ghost** [comment]
I have literally no clue how to compile on windows, and I really don't feel like using MinGW. I'm going to assume it works on windows.

Later I will test this out.

**Comment by ghost** [comment]
works on Windows

**Pull request merged**